ImageEditor/src/test/kotlin/org/wordpress/android/imageeditor/preview/PreviewImageViewModelTest.kt

```diff
@@ -20,8 +24,44 @@ class PreviewImageViewModelTest {
     }
 
     @Test
-    fun `image shown on start`() {
-        viewModel.start(TEST_IMAGE_URL)
-        assertThat(viewModel.loadImageFromUrl.value).isEqualTo(TEST_IMAGE_URL)
+    fun `data loading started on view create`() {
+        initViewModel()
+        assertThat(viewModel.uiState.value).isInstanceOf(ImageDataStartLoadingUiState::class.java)
     }
+
+    @Test
+    fun `progress bar shown on view create`() {
+        initViewModel()
+        assertThat(requireNotNull(viewModel.uiState.value).progressBarVisible).isEqualTo(true)
+    }
+
+    @Test
+    fun `progress bar shown on low res image load success`() {
+        initViewModel()
+        viewModel.onImageLoadSuccess(TEST_LOW_RES_IMAGE_URL)
+        assertThat(requireNotNull(viewModel.uiState.value).progressBarVisible).isEqualTo(true)
+    }
+
+    @Test
+    fun `progress bar hidden on low res image load failed`() {
+        initViewModel()
+        viewModel.onImageLoadFailed(TEST_LOW_RES_IMAGE_URL)
+        assertThat(requireNotNull(viewModel.uiState.value).progressBarVisible).isEqualTo(false)
+    }
+
+    @Test
+    fun `low res image success ui shown on low res image load success`() {
+        initViewModel()
+        viewModel.onImageLoadSuccess(TEST_LOW_RES_IMAGE_URL)
+        assertThat(viewModel.uiState.value).isInstanceOf(ImageInLowResLoadSuccessUiState::class.java)
+    }
+
+    @Test
+    fun `low res image failed ui shown on low res image load failed`() {
+        initViewModel()
+        viewModel.onImageLoadFailed(TEST_LOW_RES_IMAGE_URL)
+        assertThat(viewModel.uiState.value).isInstanceOf(ImageInLowResLoadFailedUiState::class.java)
+    }
+
```

Comment:
The tests are truly awesome! I'm just wondering, since we are already loading even the high res image, we might want to consider adding similar tests for the high res image, wdyt?

For example
`progress bar hidden on high res image load success`
`progress bar hidden on high res image load failed`
`high res image success ui shown on high res image load success`
`high res image failed ui shown on high res image load failed`

Wdyt?

Reply:
Another test we might want to consider adding is
`high res image success ui shown when low res image loads after high res image`